Terasology/LightAndShadow
# feat: Score hud rework

## Tasks left

- [x] Place the element correctly in the middle of the screen
- [x] Add more spacing between the team name and the score
- [x] text shadows for the team names and scores - WIP https://github.com/Terasology/LightAndShadowResources/pull/52
- [x] use the heart icon color as the font color for the "Hearts" team name - WIP https://github.com/Terasology/LightAndShadowResources/pull/52

## Purpose

This PR will hold commits regarding tasks from issue **Score HUD Element Overhaul https://github.com/Terasology/LightAndShadow/issues/131**.

## Additions included so far

- Added a UIImage in between the scores of the two teams _(ToDo: set new asset when that is committed at LASR)_
- Added team images left and right of the element 

![image](https://user-images.githubusercontent.com/48293545/88401179-11949d80-cdd2-11ea-8fff-3c4e3c504d83.png)

- Added UILabels next to the current team scores, and 
- Connected the above UILabels with `GOAL_SCORE` at `LASUtils.java`

![image](https://user-images.githubusercontent.com/48293545/88442123-f9e20700-ce1b-11ea-8b36-efba76b045e6.png)

- Added new asset, and set layout according to it - asset at https://github.com/Terasology/LightAndShadowResources/pull/47
- Removed "Score" text from the UILabels
- Move "Red" UILabel right before the heart symbol at the right end of the element

![image](https://user-images.githubusercontent.com/48293545/88442490-63aee080-ce1d-11ea-81a2-bc5808421f35.png)

- Set equal `width`s for `Black` and `Red` UILabels
- Changed UILabels `Black` to `Spades` and `Red` to `Hearts
- Added more spacing between the team name and the score
- Placed the element correctly in the middle of the screen

![image](https://user-images.githubusercontent.com/48293545/88485844-e81a7400-cf81-11ea-9e32-59d6f59a70b7.png)

## How to test

1. get LASR latest version
2. start a LAS game
3. notice the Score Element at the middle top part of the screen

![image](https://user-images.githubusercontent.com/48293545/88485857-0bddba00-cf82-11ea-8a7f-259f6d7a5766.png)
